common/src/main/java/com/microsoft/identity/common/internal/ui/webview/EmbeddedWebViewAuthorizationStrategy.java

```diff
@@ -84,15 +85,26 @@ public Future<AuthorizationResult> requestAuthorization(GenericAuthorizationRequ
         mAuthorizationRequest = authorizationRequest;
         Logger.info(methodTag,"Perform the authorization request with embedded webView.");
         final URI requestUrl = authorizationRequest.getAuthorizationRequestAsHttpRequest();
-        final Intent authIntent = buildAuthorizationActivityStartIntent(requestUrl);
+
+        String product = null;
+        String productVersion = null;
+
+        // TODO: Pretty sure this type check wouldn't be necessary, since all authorization requests that go through this path will
+        //  extend MicrosoftAuthorizationRequest. Perhaps we can change the Generic type to extend MicrosoftAuthorizationRequest rather than just AuthorizationRequest.
+        if (mAuthorizationRequest instanceof MicrosoftAuthorizationRequest) {
+            product = ((MicrosoftAuthorizationRequest) mAuthorizationRequest).getLibraryName();
```

Comment:
where are we setting the "source" of these values?

Unless i'm mistaken, i'm seeing a loop
1. This method relies on the name/version value in MicrosoftAuthorizationRequest
2. MicrosoftAuthorizationRequest relies on ProductHelper
3. ProductHelper relies on DiagnosticContext
4. DiagnosticContext values are set in setDiagnosticContextForNewThread